### DISCUSSION TITLE
Auto re-tries

### DISCUSSION THREAD

**Discussion by shikokuchuo** [wlandau/crew#169] [Q&A] [answered]
An update in `mirai`: [6a72261](https://github.com/shikokuchuo/mirai/commit/6a72261f070bbada0be872bda4d66383addcb58c) I have disabled the NNG-level auto-retry mechanism.

This doesn't break any of your tests, but it _is_ a change in behaviour so just checking in here with you. I forget if you still make use of this, or actually if this makes things easier for you.

The previous behaviour was that (assuming dispatcher) if a task crashes that the 'mirai' will remain unresolved. This behaviour isn't too helpful / transparent to a user as they would assume by default that it's still ongoing. You would need to know to check `status()` and see if the particular server was offline.

In particular, this also doesn't help with a use case like `mirai_map()`, where code like the following would have appeared to hang before the change (fortunately it's interruptible).

```r
library(mirai)
daemons(3)
m <- mirai_map(list(a = "1", b = "2", c = "q()"), function(x) eval(str2lang(x)))
m[]
```
The new behaviour explicitly returns an `errorValue`, which makes it much easier to identify crashes. What to do can then be handled by user code.

If you have a valid use case for the old behaviour, just let me know and it shouldn't be a problem to enable via an argument (I'll then delay switching the default so it is non-breaking for `crew`).

**Comment by shikokuchuo** [comment]
I've now re-introduced the option at `dispatcher()` which gains a 'retry' argument in [bd7fd6c](https://github.com/shikokuchuo/mirai/commit/bd7fd6cc3752c4c809264d68601613c588acb77a), in light of your comments at https://github.com/shikokuchuo/mirai/issues/117#issue-2321014697.

This will default to TRUE for the next release, and I haven't fully decided if this will change in the future, but `crew` can consider setting this in the `daemons()` call to be insulated from potential changes.

**Reply by wlandau** [reply, marked as answer]
Thank for doing that. After the next `mirai` release, I will set a default of `TRUE` for `crew` and allow users to set it to `FALSE` for debugging purposes (c.f. #170).
